### ISSUE TITLE
orig.ident slot/column not being add to seurat object after merge

### ISSUE THREAD

**Issue by amjass12** [satijalab/seurat#2008]
Hi all, 

I have started to encounter an issue in seurat that i have not encountered before when merging multiple seurat objects together. 

I have 11 seurat objects derived from single cell experiments (11 individual objects). They are made in to seurat objects as follows:

```
sce1<-readRDS("sce1.rds")
sce1 <- as.Seurat(sce1, project = "file1", data = "counts")

x 11 individual objects (sce1-11)
```
Once these have all been converted to seurat objects (which all worked as above without issues) I proceed to merge them. I need to run batch correction, however, because there are 11 objects. i much prefer the 'object.split' method of splitting the data before running anchor and integration methods as oppose to doing them all individually. 

For this reason, the orig.ident column is going to be very useful here! I have never encountered this issue before, however now when i run merge as follows:

`merge<merge(x=sce1, y=c(sce2,sce3,sce4,sce5,sce6,sce7,sce8,sce9,sce1,sce10),add.cell.id=c("c9","d9","a4","b4","c4","d4","e3","e4","f3","f4","g3"))`

The merge function itself works, however, there is no such column in metadata (or slot in object) called orig.ident (or anything to a similar effect). 

When viewing metadata for merge object, the renaming has worked as follows:

e4_ATCCACCAGAGCATTA.. etc etc... so the 'name' assignment has worked, however there is no such information as orig.ident:

```
Error in `[[.Seurat`(x, i, drop = TRUE) : 
  Cannot find 'orig.ident' in this Seurat object
```

I really need this information for downstream analysis and any help for a solution would be much appreciated!!

Please note, I have also tried the argument **add.cell.ids** instead of id, with the same result.

Additionally, the slot merge@active.ident does indeed give information:

"c9" 3000, "d9 4000 etc.. however, when running the following script:

`object.list <- SplitObject(lungDamage, split.by = "active.ident") ` in order to then normalize all data and findvariable features, a error is written simply stating, none of the requested variables were found "active ident"

I am guessing a work around is to normalize data and find variable features for ALL 11 objects and then merge and then make tsne's etc... i would rather be able to merge and then normalize all tofether and find variable features as per the split method... 

Many thanks!

**Comment by timoast** [comment]
Hi, can you confirm that each of the 11 objects contain `orig.ident` in the metadata? If they were converted from SingleCellExperiment then they may not.

If you have the information you want in the active idents, you add it to the object like this:

```r
merged$new.ident <- Idents(merged)
```

In your case it just seems like you need the objects in a list, and it might be easier for you to just create a new list containing the objects rather than merging and splitting them:

```r
obj.list <- list(obj1, ... , objn)
```

**Comment by amjass12** [comment]
Hi @timoast 

Thank you very much for your response, I can confirm that the metadata for the SCE object (pre and post seurat object conversion) **does not** contain the column orig.ident. 

My assumption was that this column was added during a merge.. thank you so much for clarifying!

Both creating a list as well as the merged$new.ident<-ident(merged) methods work. 

I have opted for the merge method still as the SCE data is pre-processed and contains information such as mito percentage, feature number etc.. so can filter all data sets together after merging!

May i confirm was my own sanity, that after merging (or object.list method)

that the following steps are sufficient for downstream analysis:

```
merge/list method:

merged <- SplitObject(merged, split.by = "new.ident")

for (i in 1:length(merged)) {
    merged[[i]] <- NormalizeData(merged[[i]], verbose = FALSE)
    merged[[i]] <- FindVariableFeatures(merged[[i]], selection.method = "vst", 
        nfeatures = 1000, verbose = TRUE)
}


anchors <- FindIntegrationAnchors(object.list = merged, dims = 1:20)
anchors.integrated <- IntegrateData(merged = anchors, dims = 1:20
```

etc... 

thanks! and thanks for clarifying again!
